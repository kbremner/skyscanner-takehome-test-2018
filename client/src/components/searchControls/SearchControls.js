import React from 'react';
import BpkSmallPriceAlertsIcon from 'bpk-component-icon/sm/price-alerts';
import BpkText from 'bpk-component-text';
import BpkButton from 'bpk-component-button';
import PropTypes from '../../types';
import './SearchControls.scss';

const SearchControls = ({ onFilterClick, onSortClick, onPriceAlertsClick }) => (
  <div className="search-controls">
    <BpkButton className="search-controls__filter" link onClick={onFilterClick}>Filter</BpkButton>
    <BpkButton className="search-controls__sort" link onClick={onSortClick}>Sort</BpkButton>
    <BpkButton className="search-controls__alerts" link onClick={onPriceAlertsClick}>
      <BpkSmallPriceAlertsIcon className="search-controls__alerts-icon" />
      <BpkText>Price alerts</BpkText>
    </BpkButton>
  </div>
);

SearchControls.propTypes = {
  onFilterClick: PropTypes.func.isRequired,
  onSortClick: PropTypes.func.isRequired,
  onPriceAlertsClick: PropTypes.func.isRequired,
}

export default SearchControls;
