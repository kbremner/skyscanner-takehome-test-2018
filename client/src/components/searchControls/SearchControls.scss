@import "~bpk-mixins";

.search-controls {
  padding: $bpk-spacing-sm $bpk-spacing-sm;
  display: flex;
  flex-shrink: 0;
  justify-content: space-between;
  color: $bpk-color-blue-500;
  box-shadow: $bpk-box-shadow-sm;
  z-index: 1;

  .search-controls__filter {
    flex-grow: 1;
    text-align: left;
  }

  .search-controls__sort {
    flex-grow: 4;
    text-align: left;
  }

  .search-controls__alerts {
    display: flex;
    align-items: center;

    .search-controls__alerts-icon {
      fill: $bpk-color-blue-500;
    }
  }
}