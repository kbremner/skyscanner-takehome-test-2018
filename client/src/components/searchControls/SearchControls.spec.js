import React from 'react';
import { shallow } from 'enzyme';
import SearchControls from './SearchControls';

const setup = (query = null) => {
  const filter = jest.fn();
  const sort = jest.fn();
  const priceAlerts = jest.fn();

  let wrapper = shallow(<SearchControls
    onFilterClick={filter}
    onSortClick={sort}
    onPriceAlertsClick={priceAlerts}
  />);
  if (query) {
    wrapper = wrapper.find(query);
  }

  return { filter, sort, priceAlerts, wrapper };
};

describe('SearchControls', () => {
  it('renders', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('calls provided handler when filter is clicked', () => {
    const { wrapper, filter } = setup('.search-controls__filter');
    wrapper.simulate('click');
    expect(filter).toHaveBeenCalledTimes(1);
  });

  it('calls provided handler when sort is clicked', () => {
    const { wrapper, sort } = setup('.search-controls__sort');
    wrapper.simulate('click');
    expect(sort).toHaveBeenCalledTimes(1);
  });

  it('calls provided handler when price alerts is clicked', () => {
    const { wrapper, priceAlerts } = setup('.search-controls__alerts');
    wrapper.simulate('click');
    expect(priceAlerts).toHaveBeenCalledTimes(1);
  });
});