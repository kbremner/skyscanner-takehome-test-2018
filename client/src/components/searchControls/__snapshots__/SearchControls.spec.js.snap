// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchControls renders 1`] = `
<div
  className="search-controls"
>
  <BpkButton
    className="search-controls__filter"
    destructive={false}
    disabled={false}
    featured={false}
    href={null}
    iconOnly={false}
    large={false}
    link={true}
    onClick={[MockFunction]}
    secondary={false}
    submit={false}
  >
    Filter
  </BpkButton>
  <BpkButton
    className="search-controls__sort"
    destructive={false}
    disabled={false}
    featured={false}
    href={null}
    iconOnly={false}
    large={false}
    link={true}
    onClick={[MockFunction]}
    secondary={false}
    submit={false}
  >
    Sort
  </BpkButton>
  <BpkButton
    className="search-controls__alerts"
    destructive={false}
    disabled={false}
    featured={false}
    href={null}
    iconOnly={false}
    large={false}
    link={true}
    onClick={[MockFunction]}
    secondary={false}
    submit={false}
  >
    <Component
      className="search-controls__alerts-icon"
    />
    <BpkText
      bold={false}
      className={null}
      tagName="span"
      textStyle="base"
    >
      Price alerts
    </BpkText>
  </BpkButton>
</div>
`;
